Feature/sprint 13

baha-kenpachi opened the pull request:
Критерии успешного выполнения задания

1. Переход на экран «Аудиоплеер»
2. Переход на предыдущий экран
3. UI экрана аудиоплеера
-Название трека (trackName),
-Имя исполнителя (artistName),
-Название альбома (collectionName) (если есть),
-Год релиза трека (releaseDate),
-Жанр трека (primaryGenreName),
-Страна исполнителя (country),
-Продолжительность трека в формате mm:ss (trackTimeMills),
-Кнопка добавления в плейлист,
-Кнопка добавления в избранное,
-Кнопка контроля воспроизведения,
-Прогресс воспроизведения (время под кнопкой контроля воспроизведения в формате mm:ss).
-Кроме этих полей, на экране будет отображаться обложка альбома.
4.Плейсхолдер обложки в плеере
-Если у трека нет изображения обложки, то на экране пользователь должен видеть изображение-заглушку.
-Если у трека есть изображение обложки, но оно ещё не загружено, то пользователь должен видеть изображение-заглушку.
5.Изображение обложки в плеере на 512x512bb.jpg.

Commit:
Added PlayerActivity
released onClickTrack
changed TrackData

Commit:
finished 13 sprint

Commit:
refactored some code

faritowich commented on app/src/main/java/com/example/playlistmaker/PlayerActivity.kt:
```diff
@@ -0,0 +1,84 @@
+    private lateinit var albumName: TextView
+    private lateinit var year: TextView
+    private lateinit var genre: TextView
+    private lateinit var country: TextView
```
1) Лучше давать названия, содержащие в себе сущность вью - текствью, баттон и тд. Например, tvCountry. Сейчас это некритично, но если будет большой сложный экран со сложной логикой, там могут быть разные переменные и лучше понимать сразу, где у нас вью, а где нет
2) Рекомендую почитать про библиотеку ViewBinding от гугла и использовать её - она используется почти во всех проектах.

baha-kenpachi replied:
Переделал с применением библиотеки ViewBinding

baha-kenpachi replied:
остальные активити исходя из свободного времени буду переделывать, спасибо за совет

faritowich commented on app/src/main/java/com/example/playlistmaker/PlayerActivity.kt:
```diff
@@ -0,0 +1,84 @@
+            finish()
+        }
+
+        val track = intent.getSerializableExtra("trackData") as? TrackData
```
"trackData" стоит вынести в константу и юзать тут и в SearchActivity.

baha-kenpachi replied:
вынес в константу, и некоторые другие тоже вынес и все сохранил в object Constants

faritowich commented on app/src/main/java/com/example/playlistmaker/PlayerActivity.kt:
```diff
@@ -0,0 +1,84 @@
+        }
+    }
+    fun convertPixelsToDp(px: Float, context: Context): Float {
+        return px / (context.resources.displayMetrics.densityDpi.toFloat() / DisplayMetrics.DENSITY_DEFAULT)
```
1) convertPixelsToDp - это довольно общий метод, который может пригодиться и в других местах. Его лучше вынести в отдельный файл. Назвать файл, например, Utils и в нем держать такие утильные вещи.

2) Также можно попробовать сделать такой же метод extension методом. Может, будет удобно

baha-kenpachi replied:
создал Utils

faritowich commented on app/src/main/java/com/example/playlistmaker/searchertrack/TrackData.kt:
```diff
@@ -1,10 +1,27 @@
+    val country: String
+): Serializable {
+    val trackTime: String
+        get() = SimpleDateFormat("mm:ss", Locale.getDefault()).format(trackTimeMillis)
```
Дело хорошее, можно использовать эту штуку в методе bind в TrackViewHolder

baha-kenpachi replied:
применил

baha-kenpachi replied:
@faritowich могу залить измы если больше нет замечаний

Commit:
refactored some code
fixed color on btn add and like
used ViewBinding
created object Constants
created class Utils
refactored some constants

Commit:
sprint-13:
TrackData: refactored working with nullable
Adapter: changed solution from findViewById to viewBinding
